File: Emulator/src/main/content/global/skill/summoning/familiar/npc/MossTitanNPC.java
package content.global.skill.summoning.familiar.npc;

import content.global.skill.summoning.familiar.Familiar;
import core.game.node.entity.combat.equipment.WeaponInterface;
import core.game.node.entity.player.Player;
import core.plugin.Initializable;

/**
 * The type Moss titan npc.
 */
@Initializable
public class MossTitanNPC extends ElementalTitanNPC {

    /**
     * Instantiates a new Moss titan npc.
     */
    public MossTitanNPC() {
        this(null, 7357);
    }

    /**
     * Instantiates a new Moss titan npc.
     *
     * @param owner the owner
     * @param id    the id
     */
    public MossTitanNPC(Player owner, int id) {
        super(owner, id, 5800, 12804, 20, WeaponInterface.STYLE_AGGRESSIVE);
    }

    @Override
    public Familiar construct(Player owner, int id) {
        return new MossTitanNPC(owner, id);
    }

    @Override
    public int[] getIds() {
        return new int[]{7357, 7358};
    }

}

File: Emulator/src/main/content/global/skill/summoning/familiar/npc/MossTitanNPC.kt
package content.global.skill.summoning.familiar.npc

import content.global.skill.summoning.familiar.Familiar
import core.game.node.entity.combat.equipment.WeaponInterface
import core.game.node.entity.player.Player
import core.plugin.Initializable
import org.rs.consts.Items
import org.rs.consts.NPCs

@Initializable
class MossTitanNPC
@JvmOverloads constructor(owner: Player? = null, id: Int = NPCs.MOSS_TITAN_7357) :
    ElementalTitanNPC(owner, id, 5800, Items.MOSS_TITAN_POUCH_12804, 20, WeaponInterface.STYLE_AGGRESSIVE) {
    override fun construct(owner: Player, id: Int): Familiar {
        return MossTitanNPC(owner, id)
    }

    override fun getIds(): IntArray {
        return intArrayOf(NPCs.MOSS_TITAN_7357, NPCs.MOSS_TITAN_7358)
    }
}
